ui/apps/platform/src/Containers/NetworkGraph/flows/FlowsFilterToolbar.tsx

```diff
@@ -0,0 +1,148 @@
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = prevSelection.filter((item) => item !== selection);
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        } else {
+            setFilters((prevFilters) => {
+                const prevSelection = filtersToSelections(prevFilters);
+                const newSelection = [...prevSelection, selection] as string[];
+                const newFilters = selectionsToFilters(newSelection);
+                return newFilters;
+            });
+        }
+    };
+    function handlePortsSelect(selection) {
+        setFilters((prevFilters) => {
+            const newFilters = { ...prevFilters };
+            newFilters.ports = selection;
+            return newFilters;
+        });
+    }
+
+    return (
+        <Flex>
+            <FlexItem flex={{ default: 'flex_1' }} />
+            <FlexItem>
+                <Select
+                    width="250px"
+                    variant={SelectVariant.checkbox}
+                    onToggle={onFilterDropdownToggle}
+                    onSelect={onTrafficFilterSelect}
+                    selections={selections}
+                    isOpen={isFilterDropdownOpen}
+                    placeholderText="Advanced"
+                    aria-labelledby="advanced-filters-select"
+                    isGrouped
+                    position={SelectPosition.right}
+                >
+                    <SelectGroup label="Deployment flows">
+                        <SelectOption value="anomalous">Anomalous flows</SelectOption>
+                        <SelectOption value="baseline">Baselined flows</SelectOption>
+                    </SelectGroup>
+                    <SelectGroup label="Flow directionality">
+                        <SelectOption value="ingress">Ingress (inbound)</SelectOption>
+                        <SelectOption value="egress">Egress (outbound)</SelectOption>
+                    </SelectGroup>
+                    <SelectGroup label="Protocols">
+                        <SelectOption value="TCP">TCP</SelectOption>
+                        <SelectOption value="UDP">UDP</SelectOption>
+                    </SelectGroup>
+                    <SelectGroup label="Ports">
+                        <Select
+                            className="pf-u-px-md"
+                            variant={SelectVariant.typeaheadMulti}
+                            aria-label="Select one or more days"
+                            onToggle={onTogglePortsSelect}
+                            onSelect={onSelectPorts}
+                            selections={filters.ports}
+                            isOpen={isPortsSelectOpen}
+                            placeholderText="Select ports"
+                            isCreatable
+                            createText="Select port"
+                        />
```

Comment:
I guess I could actually pass all the ports to this component and display them as an option here